src/helix-ui/elements/HXTooltipElement.js

```diff
@@ -0,0 +1,162 @@
+import { HXElement } from './HXElement';
+import { getPositionWithArrow } from '../util';
+import debounce from 'lodash/debounce';
+import shadowStyles from './_hx-tooltip.less';
+
+const tagName = 'hx-tooltip';
+const template = document.createElement('template');
+
+template.innerHTML = `
+  <style>${shadowStyles}</style>
+  <div id="container" class="position-arrow">
+    <slot></slot>
+  </div>`;
+
+export class HXTooltipElement extends HXElement {
+    static get is () {
+        return tagName;
+    }
+
+    static get observedAttributes () {
+        return [ 'open' ];
+    }
+
+    constructor () {
+        super(tagName, template);
+        this._show = this._show.bind(this);
+        this._hide = this._hide.bind(this);
+        this._toggle = this._toggle.bind(this);
+        this._setPosition = this._setPosition.bind(this);
+        this._closeOnBackgroundClick = this._closeOnBackgroundClick.bind(this);
+    }
+
+    connectedCallback () {
+        if (!this.position) {
+            this.position = 'top';
+        }
+
+        this.initialPosition = this.position;
+        this.$upgradeProperty('open');        
+        this.$defaultAttribute('role', 'tooltip');
+
+        if (this.id) {
+            this._target = document.querySelector('[data-tooltip=' + this.id + ']');
+        } else {
+            return;
+        }
+        this._connectHandlers(this.trigger);
+    }
+
+    disconnectedCallback () {
+        if (!this._target) {
+            return;
+        }
+        this._destoryAllHandlers();
+    }
+
+    attributeChangedCallback (attr, oldValue, newValue) {
+        this.setAttribute('aria-hidden', newValue !== '');
+    }
+
+    _hide () {
+        if (this._showTimer) {
+            clearTimeout(this._showTimer);
+        }
+        this._hideTimer = setTimeout(() => {
+            this.open = false;
+        }, 1600);
+    }
+
+    _show () {
+        if (this._hideTimer) {
+            clearTimeout(this._hideTimer);
+        }
+        this._showTimer = setTimeout(() => {
+            this.open = true;
+        }, 500);
+    }
+
+    _toggle () {
+        this.open = !this.open;
+    }
+
+    _closeOnBackgroundClick (event) {
+        if (this._isBackground(event)) {
+            this.open = false;
+        }
+    }
+
+    _connectHandlers (type) {
```

Comment:
Given that this is a private function, and we have access to `this.trigger` (or `this.triggerEvent`) we probably don't need to pass the event type as an argument to the function.

Reply:
yeah good catch